packages/@ourworldindata/grapher/src/entitySelector/EntitySelector.tsx

```diff
@@ -99,6 +171,69 @@ export class EntitySelector extends React.Component<{
         return this.manager.entitySelectorState.mostRecentlySelectedEntityName
     }
 
+    @computed private get sortConfig(): SortConfig {
+        return (
+            this.manager.entitySelectorState.sortConfig ??
+            this.defaultSortConfig
+        )
+    }
+
+    @computed private get table(): OwidTable {
+        return this.manager.tableForSelection
+    }
+
+    @computed private get sortColumns(): CoreColumn[] {
+        const activeSlugs = this.manager.activeColumnSlugs ?? []
+
+        const sortColumns = activeSlugs
+            .map((slug) => this.table.get(slug))
+            .filter(
+                (column) => column.hasNumberFormatting && !column.isProjection
+            )
+
+        return [this.table.entityNameColumn, ...sortColumns]
+    }
```

Comment:
Very bad news here, sadly:
This innocent-looking function can take 3+ seconds to run, and is render-blocking.

To repro, go to http://staging-site-entity-selector-sort/grapher/excess-deaths-cumulative-economist-single-entity in a narrow viewport (such that the entity selector doesn't render yet), then resize the window to make the selector show up. The result is that grapher is unresponsive for 6+ seconds, as `LineChart.transformTableForSelection` is running.

I'm a bit confused how this isn't the case on live currently. For some reason, on live, `chartInstance.transformTableForSelection` was called once, and now it's being called 3 times.

I can try to dig into this a bit - for example, to look for perf improvements in `LineChart.transformTableForSelection`...

Reply:
Fix for `transformTableForSelection` perf at #3486 🌻

Reply:
Nice! Thanks a lot for looking into this :)